test/from-draft7/to-2019-09/dependencies-to-dependentSchemas.json

```diff
@@ -0,0 +1,32 @@
+[
+    {
+        "title": "`dependencies` has properties whose subschema has type object",
+        "from": {
+            "dependencies": {
+                "foo": {
+                    "minProperties": 2
+                }
+            }
+        },
+        "to": {
+            "dependentSchemas": {
+                "foo": {
+                    "minProperties": 2
+                }
+            }
+        }
+    },
+    {
+        "title": "`dependencies` has properties whose subschema does not have type object",
+        "from": {
+            "dependencies": {
+                "foo": [ "bar", "baz" ]
+            }
+        },
+        "to": {
+            "dependencies": {
```

Comment:
Here, you should transform to `dependentRequired`?

Reply:
Here too.. I should include this case..?

Reply:
Same as above. Not sure I'm following. Even on Alterschema right now, this test does not pass, as the transformation does take place (as expected?)

<img width="1624" alt="Screenshot 2024-07-23 at 9 53 33 PM" src="https://github.com/user-attachments/assets/75e3d1d0-9bbd-44ce-8bce-f92c85ce6d1e">

Reply:
@jviotti  it is to show that the rule given for arrays the transform should not happen for objects and vise versa, 
I have split the transform, and for empty objects and arrays: instead of including cleanup in rules why not let teh engine to cleanup in the end.
OR should I include cleanup in rules?

Reply:
> it is to show that the rule given for arrays the transform should not happen for objects and vise versa

I don't think you can define those tests, as you cannot test rules in isolation. Our tests never apply to specific rules, but are end to end tests that cover all the rules together.